x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/IlmHealthIndicatorService.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.ilm;
+
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.health.HealthIndicatorDetails;
+import org.elasticsearch.health.HealthIndicatorResult;
+import org.elasticsearch.health.HealthIndicatorService;
+import org.elasticsearch.health.SimpleHealthIndicatorDetails;
+import org.elasticsearch.xpack.core.ilm.IndexLifecycleMetadata;
+import org.elasticsearch.xpack.core.ilm.OperationMode;
+
+import java.util.Map;
+
+import static org.elasticsearch.health.HealthStatus.GREEN;
+import static org.elasticsearch.health.HealthStatus.YELLOW;
+import static org.elasticsearch.health.ServerHealthComponents.DATA;
+
+public class IlmHealthIndicatorService implements HealthIndicatorService {
+
+    public static final String NAME = "ILM";
+
+    private final ClusterService clusterService;
+
+    public IlmHealthIndicatorService(ClusterService clusterService) {
+        this.clusterService = clusterService;
+    }
+
+    @Override
+    public String name() {
+        return NAME;
+    }
+
+    @Override
+    public String component() {
+        return DATA;
+    }
+
+    @Override
+    public HealthIndicatorResult calculate() {
+        var ilmMetadata = clusterService.state().metadata().custom(IndexLifecycleMetadata.TYPE, IndexLifecycleMetadata.EMPTY);
+        if (ilmMetadata.getPolicyMetadatas().isEmpty()) {
+            return createIndicator(GREEN, "No policies configured", createDetails(ilmMetadata));
+        } else if (ilmMetadata.getOperationMode() != OperationMode.RUNNING) {
+            return createIndicator(YELLOW, "Ilm is not running", createDetails(ilmMetadata));
```

Comment:
I think this can be more fine-grained, ILM has two non-running modes, `STOPPING` and `STOPPED`, so the message could be changed to `ILM is currently stopping` and `ILM is not running` or `ILM is stopped` for the different statuses to indicate the difference.